Abhi-AnimeX/app/src/main/java/com/abhijith/animex/ui/screens/animelist/viewmodel/AnimeListViewModel.kt

```diff
@@ -1,34 +1,55 @@
 package com.abhijith.animex.ui.screens.animelist.viewmodel
 
+import android.util.Log
 import androidx.lifecycle.ViewModel
-import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.viewModelScope
-import com.abhijith.animex.domain.model.AnimeItem
-import com.abhijith.animex.domain.usecases.GetAnimeListUseCase
+import com.abhijith.animex.domain.models.AnimeItem
+import com.abhijith.animex.domain.usecases.GetSeasonalAnimeListUseCase
 import com.abhijith.animex.ui.screens.animelist.AnimeListUiState
-import kotlinx.coroutines.delay
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.SupervisorJob
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.collectLatest
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.plus
+import javax.inject.Inject
+
+@HiltViewModel
+class AnimeListViewModel @Inject constructor(
+    private val seasonalAnimeListUseCase: GetSeasonalAnimeListUseCase
+) :
+    ViewModel() {
+
+    private val tag = AnimeListViewModel::class.java.simpleName
 
-class AnimeListViewModel(private val getAnimeListUseCase: GetAnimeListUseCase) : ViewModel() {
     private val _itemsUiState = MutableStateFlow<AnimeListUiState>(AnimeListUiState.Loading)
     val itemsUiState: StateFlow<AnimeListUiState> = _itemsUiState.asStateFlow()
 
     private val _navigationEvent = MutableStateFlow<AnimeItem?>(null)
     val navigationEvent: StateFlow<AnimeItem?> = _navigationEvent.asStateFlow()
 
+    private val job = SupervisorJob()
+    private val scope = viewModelScope + job
```

Comment:
why this?

Reply:
just in case in future there are more calls on this page, i wanted to make sure the sibling coroutines doesn't stop the main ones which is running already